datasources/beeper/beeper.go

```diff
@@ -0,0 +1,982 @@
+					eventID = toString(values[eventIDIdx])
+				}
+
+				if roomIDIdx >= 0 && roomIDIdx < len(values) {
+					roomID = toString(values[roomIDIdx])
+				}
+
+				if timestampIdx >= 0 && timestampIdx < len(values) {
+					timestamp = toTimestamp(values[timestampIdx])
+				} else {
+					// Default to current time if no timestamp
+					timestamp = time.Now()
+				}
+
+				// Try to get text content from either message JSON or text_content field
+				if textContentIdx >= 0 && textContentIdx < len(values) {
+					messageText = toString(values[textContentIdx])
+				}
+
+				if messageText == "" && messageIdx >= 0 && messageIdx < len(values) {
+					messageJSON := toString(values[messageIdx])
+					// Try to extract text from JSON if possible
+					if strings.Contains(messageJSON, "\"body\":") {
+						// This is a very simple JSON extraction - in a real implementation
+						// you would properly parse the JSON
+						parts := strings.Split(messageJSON, "\"body\":")
+						if len(parts) > 1 {
+							bodyPart := parts[1]
+							if strings.Contains(bodyPart, "\"") {
+								bodyPart = strings.Split(bodyPart, "\"")[1]
+								messageText = bodyPart
+							}
+						}
+					}
+				}
+
+				if senderContactIDIdx >= 0 && senderContactIDIdx < len(values) {
+					senderContactID = toString(values[senderContactIDIdx])
+				}
+
+				if isSentByMeIdx >= 0 && isSentByMeIdx < len(values) {
+					isSentByMeVal := values[isSentByMeIdx]
+					switch v := isSentByMeVal.(type) {
+					case int64:
+						isSentByMe = v > 0
+					case bool:
+						isSentByMe = v
+					default:
+						isSentByMe = false
+					}
+				}
+
+				// Use threadID/roomID as chat name
+				chat := roomID
+				if chat == "" {
+					chat = "Beeper Chat"
+				}
+
+				// Use sender if available
+				sender := senderContactID
+				if sender == "" {
+					if isSentByMe {
+						sender = "Me"
+					} else {
+						sender = "Unknown"
+					}
+				}
+
+				// Create metadata map
+				metadata := timeline.Metadata{
+					"Source":     "Beeper",
+					"ChatApp":    "beeper",
+					"Chat":       chat,
+					"Sender":     sender,
+					"MessageID":  eventID,
+					"IsSentByMe": isSentByMe,
+				}
+
+				// Add all columns to metadata
+				for i, col := range columns {
+					metadata[col] = toString(values[i])
+				}
+
+				// Create item
+				item := &timeline.Item{
+					Classification: timeline.ClassMessage,
+					Timestamp:      timestamp,
+					Content: timeline.ItemData{
+						Data: timeline.StringData(messageText),
+					},
+					Metadata: metadata,
+				}
+
+				// Create entity if we have a sender
+				var entity *timeline.Entity
+				if sender != "" && sender != "Unknown" {
+					entity = &timeline.Entity{
+						Name: sender,
+						Attributes: []timeline.Attribute{
+							{
+								Name:     "beeper_contact_id",
+								Value:    senderContactID,
+								Identity: true,
+							},
+							{
+								Name:     "chat_platform",
+								Value:    "beeper",
+								Identity: false,
```

Comment:
I'm not sure this attribute provides any particular value (same with the other occurrence of it)